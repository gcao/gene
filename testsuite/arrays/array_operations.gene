#!/usr/bin/env gene run
# TEST: Array operations

# Array creation
(var arr [1 2 3 4 5])
(print "array: " arr)

# Array indexing
(print "arr[0] = " (arr 0))
(print "arr[2] = " (arr 2))
(print "arr[4] = " (arr 4))

# Array length
(print "length: " (.length arr))

# Array methods
(var arr2 [])
(.push arr2 10)
(.push arr2 20)
(.push arr2 30)
(print "after push: " arr2)

(print "pop: " (.pop arr2))
(print "after pop: " arr2)

# Array concatenation
(var arr3 [1 2 3])
(var arr4 [4 5 6])
(print "concat: " (.concat arr3 arr4))

# Array slicing
(var arr5 [10 20 30 40 50])
(print "slice(1, 3): " (.slice arr5 1 3))
(print "slice(2): " (.slice arr5 2))