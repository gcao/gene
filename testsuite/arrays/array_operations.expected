array: [1 2 3 4 5]
arr[0] = 1
arr[2] = 3
arr[4] = 5
length: 5
after push: [10 20 30]
pop: 30
after pop: [10 20]
concat: [1 2 3 4 5 6]
slice(1, 3): [20 30]
slice(2): [30 40 50]